packages/SwingSet/tools/bundleTool.js

```diff
@@ -172,10 +172,11 @@ export const makeBundleCache = (wr, cwd, readPowers, opts) => {
   const loaded = new Map();
   /**
    * @param {string} rootPath
-   * @param {string} targetName
+   * @param {string} [targetName]
    * @param {(message: *) => void} [log]
    */
   const load = async (rootPath, targetName, log = console.debug) => {
+    targetName ||= pathAmbient.basename(rootPath, '.js');
```

Comment:
the `||=` is why this PR is stacked on https://github.com/Agoric/agoric-sdk/pull/7073

@dckc I know you'll have thoughts about `pathAmbient`. It was a hassle to wire it through the functions and they don't have types so that would be lost. I figure the module is like a closure in which an authority is available so this is not a violation of ocap.

I figure at some point we'll have things like Node's `path` not importable. Until then this feels pragmatic.

Reply:
> It was a hassle to wire it through the functions ...

Adding `basename` to `readPowers` isn't straightforward?

> I figure the module is like a closure in which an authority is available so this is not a violation of ocap.

Hm. Exporting functions that close over ambient authority such as `path.basename` is one of the few crisp prohibitions we have managed to articulate (or... I thought we had managed...):

> No modules should export powerful objects (for example, objects that close over `fs` or `process.env`).
 -- [OCap Discipline](https://github.com/Agoric/agoric-sdk/wiki/OCap-Discipline)

Reply:
> Adding basename to readPowers isn't straightforward?

Adding it to `makeReadPowers` isn't because that's in Endo. This works,
```js
  const readPowers = {
    ...makeReadPowers({ fs, url, crypto }),
    basename: fs.basename,
  };
```
But it's confusing the `makeReadPowers` doesn't make `readPowers. And the type information is lost. The net cost of repeating every `fs` function that we need strikes me as greater than its benefit.

> No modules should export powerful objects (for example, objects that close over fs or process.env).

I don't think we fully follow that edict. A search of `process.env` shows 77 results in 40 files. Not all are functions closing over it but I see 10-20 that are.

I'll remove pathAmbient to get this PR in but I think the ocap coding guidelines could use some refinement.

Reply:
> > No modules should export powerful objects (for example, objects that close over fs or process.env).
> 
> I don't think we fully follow that edict.

Indeed;  #2160 represents the technical-debt to clean that up.

> And the type information is lost.

That seems straightforward to address.

> The net cost of repeating every `fs` function that we need strikes me as greater than its benefit.

It is unfortunate that node's `fs` API is awkward w.r.t. the principle of least authority. I hope we can use something shaped like this more widely...

https://github.com/Agoric/agoric-sdk/blob/010105a0e4471a81a97d567bf04cb48753e0ec37/packages/SwingSet/tools/bundleTool.js#L9-L31

Reply:
> Adding it to `makeReadPowers` isn't because that's in Endo.

Ah... I missed that point earlier. Now I appreciate why it's awkward.

Here's hoping I can sync up with endo on POLA-shaped IO soonish. Thanks for ef90a4f meanwhile.